### PR TITLE
C#/Java: Re-factor the `isSupported` predicate.

### PR DESCRIPTION
**Pull request by michaelnebel**
It turns out that all uses of `isSupported` also includes logic to cater for neutrals. In this PR we put this into the `isSupported` predicate itself.
Thank you @starcke!

### CODE REVIEW

**Comment by michaelnebel** [comment]
DCA looks good, except there seems to be an issue with a C# project that doesn't build anymore (this is also an issue in another DCA run).

**Review by aschackmull — DISMISSED** [review]

File: csharp/ql/src/Telemetry/ExternalApi.qll [hunk 1]
```diff
@@ -104,8 +105,14 @@ class ExternalApi extends DotNet::Callable {
   pragma[nomagic]
   predicate isSink() { sinkNode(this.getAnInput(), _) }
 
+  /** Holds if this API is a known neutral. */
+  pragma[nomagic]
+  predicate isNeutral() { this instanceof FlowSummaryImpl::Public::NeutralCallable }
+
   /** Holds if this API is supported by existing CodeQL libraries, that is, it is either a recognized source or sink or has a flow summary. */
```
Comment (mbg) [inline csharp/ql/src/Telemetry/ExternalApi.qll:112]: Should this comment be updated to reflect the additional condition?

**Review by mbg — APPROVED** [review]
LGTM aside from one minor comment.

File: csharp/ql/src/Telemetry/ExternalApi.qll (hunk 1 above)
Comment (michaelnebel) [reply csharp/ql/src/Telemetry/ExternalApi.qll:112]: Yes, definitely! Thank you!

**Review by mbg — APPROVED** [review]

**Pull request merged**